src/parameters.h

```diff
@@ -16,7 +16,7 @@
 class Parameters
 {
 private:
-	Parameters(ContextFrame&& frame);
+	Parameters(ContextFrame&& frame, const Location& loc);
```

Comment:
Parameters gained the `valid` and `valid_required` type-and-presence checking functions, so that you can say
```
if (!parameters.valid("somenumber", Value::Type::NUMBER)) {
    handle error somehow
}
```
and it will spit out a warning.  To spit out a warning with a file name and line number, it needs a documentPath and a Location.  Note that the creation of the Parameters already got the location as an argument; this just retains it for use in subsequent checks.